From f3b72eeeb3a6fc1c3579970f3caad75eb9d9fcfc Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Wed, 31 Oct 2018 17:46:15 +0800
Subject: [PATCH 4962/5242] MLK-20142-2 PCI: imx: refine codes in the
 initialization

commit  d3bb4174f1a3122cf4b8cb9fd46164248292e4f2 from
https://source.codeaurora.org/external/imx/linux-imx.git

- replace the sleep by the udelay, since it would be used
in the no_irq_suspend/resume callbacks.
- aligned the retries to the PHY_PLL_LOCK_WAIT_MAX_RETRIES

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 771cc7c..9aae0d7 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -666,8 +666,7 @@ static int imx7d_pcie_wait_for_phy_pll_lock(struct imx_pcie *imx_pcie)
 		if (val & IMX7D_GPR22_PCIE_PHY_PLL_LOCKED)
 			return 0;
 
-		usleep_range(PHY_PLL_LOCK_WAIT_USLEEP_MIN,
-			     PHY_PLL_LOCK_WAIT_USLEEP_MAX);
+		udelay(PHY_PLL_LOCK_WAIT_USLEEP_MIN);
 	}
 
 	dev_err(dev, "PCIe PLL lock timeout\n");
@@ -682,7 +681,8 @@ static int imx8_pcie_wait_for_phy_pll_lock(struct imx_pcie *imx_pcie)
 	struct device *dev = pci->dev;
 
 	if (imx_pcie->variant == IMX8MM) {
-		for (retries = 0; retries < 100; retries++) {
+		for (retries = 0; retries < PHY_PLL_LOCK_WAIT_MAX_RETRIES;
+				retries++) {
 			tmp = readl(imx_pcie->phy_base + PCIE_PHY_CMN_REG75);
 			if (tmp == PCIE_PHY_CMN_REG75_PLL_DONE)
 				break;
@@ -690,7 +690,8 @@ static int imx8_pcie_wait_for_phy_pll_lock(struct imx_pcie *imx_pcie)
 		}
 	} else if (imx_pcie->variant == IMX8QXP
 			|| imx_pcie->variant == IMX8QM) {
-		for (retries = 0; retries < 100; retries++) {
+		for (retries = 0; retries < PHY_PLL_LOCK_WAIT_MAX_RETRIES;
+				retries++) {
 			if (imx_pcie->hsio_cfg == PCIEAX1PCIEBX1SATA) {
 				regmap_read(imx_pcie->iomuxc_gpr,
 					    IMX8QM_CSR_PHYX2_OFFSET + 0x4,
-- 
1.7.9.5

